### DISCUSSION TITLE
Accessing LLC

### DISCUSSION THREAD

**Discussion by MathanMK** [ChampSim/ChampSim#354] [Q&A] [answered]
Why is LLC initialized as L1D's lower level's lower level rather than L2C'c lower level. Also If I want to send a packet directly to LLC, how can I achieve it?
    # LLC
    cache_name = caches[caches[cpu['L1D']]['lower_level']]['lower_level']
    if cache_name != 'DRAM':
        caches[cache_name] = ChainMap(caches[cache_name], default_llc.copy())

How to modify the code, so that LLC, can be accessed as L2C's lower level?

**Comment by ngober** [comment, marked as answer]
The configuration step in ChampSim tries to make as many good defaults as it can. One guess is that the system has exactly three levels of cache. In this, the LLC is two steps down the hierarchy from the L1D. Another is that the last-level cache is shared among all cores. If these two guesses happen to refer to the same cache, we can merge the assumptions we make. However, we don't name it the "last-level cache" by default. We assign the "third-level cache" to have a certain size, and that's what's happening here.

Regarding your second question, there is no way to send a packet directly to the LLC. We cannot make memory requests materialize out of nothing. The closest approximation of this would be a prefetcher in the LLC.
